src/measure_theory/integration.lean

```diff
@@ -1387,3 +1419,20 @@ measure.of_measurable_apply s hs
 end lintegral
 
 end measure_theory
+
+open measure_theory measure_theory.simple_func
+/-- To prove something for an arbitrary measurable function into `ennreal`, it sufficient to show
+that the property holds for (multiples of) characteristic functions and is closed under addition
+and supremum of increasing sequences of functions. -/
+theorem measurable.ennreal_induction {α} [measurable_space α] {P : (α → ennreal) → Prop}
+  (h_ind : ∀ (c : ennreal) ⦃s⦄, is_measurable s → P (indicator s (λ _, c)))
+  (h_sum : ∀ ⦃f g⦄, measurable f → measurable g → P f → P g → P (f + g))
```

Comment:
sometimes, proving that a property is stable under sum is much easier when the functions have disjoint support. I think your induction principle still holds under this weaker assumption, right?

Reply:
It does. I was considering only adding it when I ever needed it, but I can add it now if you foresee that this will be useful.

I was a little hesitant, since there are many other conditions you can put on `f` and `g` (for example: `f` is a simple function and `g` is an indicator function with disjoint support and the intersection of their images is a subset of `{0}`), and I'm not sure which of those will actually simplify arguments.